lib/Config.php

```diff
@@ -44,13 +55,24 @@ private function __construct()
         $this->store = $this->config->getConfigItem(self::STORE, null);
         $this->tables = $this->config->getArray('tables', []);
         $this->mode = $this->config->getValueValidate(self::MODE, ['PROXY', 'IDP', 'SP', 'MULTI_IDP'], 'PROXY');
+        $this->apiWriteEnabled = $this->config->getBoolean(self::API_WRITE_ENABLED, false);
+        if ($this->apiWriteEnabled) {
+            $this->apiWriteUsername = $this->config->getString(self::API_WRITE_USERNAME);
+            if (null === $this->apiWriteUsername || empty(trim($this->apiWriteUsername))) {
+                throw new Exception('Username for API write cannot be empty');
+            }
+            $this->apiWritePassword = $this->config->getString(self::API_WRITE_PASSWORD);
+            if (null === $this->apiWritePassword || empty(trim($this->apiWritePassword))) {
```

Comment:
```suggestion
            if (empty(trim($this->apiWriteUsername))) {
                throw new Exception('Username for API write cannot be empty');
            }
            $this->apiWritePassword = $this->config->getString(self::API_WRITE_PASSWORD);
            if (empty(trim($this->apiWritePassword))) {
```
[Configuration::getString()](https://github.com/simplesamlphp/simplesamlphp/blob/da0df94ab05da52ce27ab7c2a360567046bf03b2/lib/SimpleSAML/Configuration.php#L719) never returns `null`, so the `null` check is redundant.